e2e/controls.py

```diff
@@ -114,3 +166,128 @@ def locate_by_label(self, label: str) -> Locator:
 
     def locate_by_value(self, value: str) -> Locator:
         return self.loc.locator("input[type=radio][value={value}]")
+
+class SelectInput(SimpleInput):
+    def __init__(self, page: Page, inputId: str):
+        super().__init__(page, f"select#{inputId}.shiny-bound-input")
+
+    # TODO
+    # def get_options(self):
+
+    def get_selected(self):
+        return self.loc.locator("option[selected]").inner_text()
+
+    def select_option(self, value: str):
+        self.loc.select_option(value)
+
+class SelectizeInput():
+    def __init__(self, page: Page, inputId: str):
+        self.page = page
+        self.loc = page.locator(f"//*[@id='{inputId}']//ancestor::div[contains(@class, 'shiny-input-container')]")
+
+    def get_selected_items(self) -> List:
+        selected_items = self.loc.locator(".selectize-input .item")
+        mylist = []
+        for i in range(selected_items.count()):
+            mylist.append(selected_items.nth(i).inner_text())
+
+        return mylist
+
+    def select_option(self, value: str):
+        self.loc.locator(".selectize-input").click()
+
+        dropdown = self.loc.locator(".selectize-dropdown")
+        expect(dropdown).to_be_visible()
+
+        self.loc.locator(f".selectize-dropdown-content .option[data-value='{value}']").click()
+
+        self.loc.locator(f".selectize-input .item[data-value='{value}']")
+
+        # click outside the dropdown box to close the dropdown menu
+        self.page.keyboard.press("Escape")
+
+    #TODO
+    #def remove_option(self):
+
+
+class FileInput():
+    def __init__(self, page: Page, inputId: str):
+        self.loc = page.locator(f"//*[@id='{inputId}']//parent::div[contains(@class, 'shiny-input-container')]")
+
+    def upload_file(self, fileName: str) -> None:
+        browse = self.loc.locator("//span[contains(@class, 'btn-file')]")
+        #TODO: Enhancement: Check PurePath option to upload files
+        browse.set_input_files(files=[f"e2e/data-files/{fileName}"])
```

Comment:
This feels too specific--I think `upload_file` should take `filenames: list[str]` as a parameter and pass it straight through to `set_input_files`, and make it the responsibility of the caller to point to e2e/data-files/ if that is what they want to do.